src/tests/stuf/test_message.py

```diff
@@ -179,3 +183,22 @@ def test_get_elm_attr(self):
 
         e = stuf.get_elm_value_by_path("elm3", ".//elm3sub//sub[@x='5']")
         self.assertEqual(e, None)
+
+    def test_create_elm(self):
+        stuf_message = StufMessage(self.msg)
+
+        elm = 'elm4 elm5 elm6'
+
+        stuf_message.create_elm(elm)
+        stuf_message.set_elm_value(elm, 'value of new elm6')
+
+        self.assertEqual('value of new elm6', stuf_message.get_elm_value(elm))
+
+        # Already exists. Original value should be returned
+        stuf_message.create_elm('elm1')
+        self.assertEqual('value', stuf_message.get_elm_value('elm1'))
+
+        # Create element in root
+        stuf_message.create_elm('elm7')
+        stuf_message.set_elm_value('elm7', 'elm7value')
+        self.assertEqual('elm7value', stuf_message.get_elm_value('elm7'))
```

Comment:
Can't you include a pretty printed output of the resulting xml? That makes even more clear what the result of this method is. Moreover it uses the pretty_print method which took me a while to discover but is easy for debugging

Reply:
I've added an example for the first test case.